lib/rules/header.js

```diff
@@ -51,13 +51,18 @@ module.exports = function(context) {
 
     return {
         Program: function(node) {
-            var leadingComments;
-            if (node.body.length) {
-                leadingComments = context.getComments(node.body[0]).leading;
-            } else {
-                leadingComments = context.getComments(node).leading;
+            function excludeShebangs(comments) {
+                return comments.filter(function(comment) { return comment.type !== "Shebang"; });
+            }
+
+            function getLeadingComments() {
+                return node.body.length ?
+                    context.getComments(node.body[0]).leading :
+                    context.getComments(node).leading;
```

Comment:
These functions will get created on every Program call. Could you move them to the root of this file? (Obviously the node will have to get passed in to getLeadingComments)

Reply:
No problem, done